plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/steps/payment-methods-selection/index.tsx

```diff
@@ -114,9 +117,25 @@ export default function PaymentMethodsSelection() {
 	}, [ recommendedPaymentMethods, isExpanded, initialVisibilityMap ] );
 
 	const savePaymentMethodsState = (
-		state: Record< string, boolean >
+		state: Record< string, boolean >,
+		changedMethodId?: string
 	): Promise< void > => {
 		const saveUrl = currentStep?.actions?.save?.href;
+
+		// Store the previous state for potential rollback
+		const previousState = paymentMethodsState;
```

Comment:
It may be better to create a shallow copy of `paymentMethodsState`, as `previousState` currently holds a reference to the original object. Once it’s updated, rolling back becomes impossible.

```suggestion
		const previousState = { ...paymentMethodsState };
```

Reply:
Good call! Fixed in https://github.com/woocommerce/woocommerce/pull/60602/commits/819850399fffc16548b82cf819ef9bd0878ea63a